### ISSUE TITLE
Two feature recommend

### ISSUE THREAD

**Issue by ayanmw** [Supervisor/supervisor#1290]
I want two feature support:

- [ ] First is simple regex support about the name 

like `supervisorctl stop *server*:*` 

- [ ] Second is I want `stop all` will do stop works in order by stopped 

like prog1 prog2 ,priority is 1,2 ,`stop all` will do work like:

1. stop prog2
2. makesure prog2 is stopped
3. stop prog1 
4. makesure prog2 is stopped

- [ ]  `restart all` will do the same thing like above, stopped and then `start all`

If supervisor is already support above feature ,please point me that! thanks!

**Comment by ayanmw** [comment]
My config is like  prog2 depends on prog1 , but when `stop all` ,the prog1 may be stopped but prog2 is still in stopping ,that will cause some problem!

**Comment by mnaberez** [comment]
> First is simple regex support about the name

Duplicate of #655 

> Second is I want stop all will do stop works in order by stopped
> restart all will do the same thing like above, stopped and then start all

Duplicate of #122 (not an exact duplicate by the description but that's the issue we are using to track requests for program dependencies)
